examples/hosts/app-integration/external-data/README.md

```diff
@@ -6,15 +6,68 @@ Please note that the ideas explored here are experimental and under development.
+
+1. POST `/register-for-webhook`: Register's the sender's URL (in this case the Customer Service's URL) to receive notifications when the external task-list data changes.
+2. GET `/fetch-tasks`: Fetches the task list from the external data store.
+3. POST `/set-tasks`: Updates external data store with new tasks list (complete override).
+4. POST `/debug-reset-task-list`: Resets the external data to its original contents.
+
+Find the details of the API in the [External Data Service README](./src/mock-external-data-service/README.md)
+
+**Customer Service**
+
+Next we need a customer service that functions as the intermediary between the External Data Service and the Fluid Service. This server is responsible for authenticating to the external service on the customer's behalf. It registers to the External Data Service webhooks and listens for incoming changes. It also acts as a translation layer, tasnlating to and from the External Data Service expected format and the Fluid Service's expected format.
+
+In this example, the Customer Service contains the following endpoints:
+
+1. POST `/register-for-webhook`: Register's the sender's URL (in this case, the Fluid Service's URL) to receive notifications when the external task-list data changes.
+2. POST `/echo-external-data-webhook`: "Echoes" the external data services data update notifications to our own webhook subscribers. This data will be forwarded to our own subscribers.
+
+Find the details of the API in the [Customer Service README](./src/mock-customer-service/README.md).
+
+Next we come to how the Fluid collaboration session and clients will consume the APIs above and render the data to the screen. This example repo explores two routes to accomplish this. One is the Echo Webhook Pattern and the other is the Bot Pattern. Both are documented in more detail below.
 
 ### Echo Webhook Pattern
 
-TODO: Document the echo-webhook pattern and how it is surfaced in the example code.
+<img width="1356" alt="image" src="https://user-images.githubusercontent.com/6777404/216415477-14d0b193-29c9-48e5-8b6b-0a549a5dde58.png">
+
+<img width="1404" alt="image" src="https://user-images.githubusercontent.com/6777404/216417448-2a43db3e-12a2-48a6-b6d0-a6c4a95e27d1.png">
+
+<img width="1374" alt="image" src="https://user-images.githubusercontent.com/6777404/216417779-12861504-7909-489c-b7a2-4d75814a396f.png">
+
+Since the Customer Service is registered to the webhooks and listening for incoming changes, the clients do not have a way to know that there has been a change upstream.
+
+The "echo webhook" pattern accomplishes this by having the Fluid service subscribe to "echo webhooks" in the Customer Service, listening for the webhook notifications there, and then sending a Signal to the clients to alert them of this upstream change.
```

Comment:
I would maybe reword this to something like "... the Fluid service subscribe to notifications from the Customer Service when it receives webhook notifications from the external data provider...". "Echo Webhook" is the name we've given the pattern here, but I think being more descriptive is probably useful.

Reply:
Same with subsequent references to "echo webhooks" - I think describing the scenarios in terms of subscription / notifications will probably be easier readers to understand.